src/Symfony/Component/Mailer/Exception/InvalidDsnException.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Mailer\Exception;
+
+use Symfony\Component\Mailer\Bridge;
+use Symfony\Component\Mailer\Transport\Dsn;
+
+/**
+ * @author Konstantin Myakshin <molodchick@gmail.com>
+ */
+class InvalidDsnException extends LogicException
+{
+    public static function unsupportedScheme(Dsn $dsn): self
+    {
+        return new self(sprintf('The "%s" scheme is not supported for mailer "%s".', $dsn->getScheme(), $dsn->getHost()));
+    }
+
+    public static function unsupportedHost(Dsn $dsn): self
```

Comment:
We are using the Doctrine's style of doing exceptions. You need to create 2 different exception classes instead.